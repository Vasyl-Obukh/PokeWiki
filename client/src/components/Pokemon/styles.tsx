import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  padding-bottom: 50px;
  flex-direction: column;
  align-items: center;
`;

export const Image = styled.img`
  width: 275px;
  height: 275px;
  border-radius: 50%;
  background-color: #f0f0f0;
  box-shadow: 0 0 5px darkslategray;
`;

export const Info = styled.div`
  display: grid;
  margin-top: 25px;
  grid-template-columns: auto auto;
  grid-column-gap: 5px;
  grid-row-gap: 10px;
  justify-content: center;
`;

export const InfoKey = styled.div`
  font-weight: bold;
  text-transform: capitalize;
  text-align: right;
  
  &::after {
    content: ':';
  }
`;

export const InfoValue = styled.div`
  font-style: italic;
  font-weight: ${props => props.primary ? 'bold' : 'normal'};
  text-transform: ${props => props.primary ? 'uppercase' : 'none'};
`;

export const ValueItem = styled.span`
  padding: 5px;
  
  &:not(:first-child) {
    border-left: 1px solid gray;
  }
`;