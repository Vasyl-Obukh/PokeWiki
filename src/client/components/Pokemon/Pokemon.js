import React, { Fragment } from 'react';
import styled from 'styled-components';

const PokemonWrapper = styled.div`
  display: flex;
  padding-bottom: 50px;
  flex-direction: column;
  align-items: center;
`;

const PokemonImage = styled.img`
  width: 275px;
  height: 275px;
  border-radius: 50%;
  background-color: #f0f0f0;
  box-shadow: 0 0 5px darkslategray;
`;

const PokemonInfo = styled.div`
  display: grid;
  margin-top: 25px;
  grid-template-columns: auto auto;
  grid-column-gap: 5px;
  grid-row-gap: 10px;
  justify-content: center;
`;

const InfoKey = styled.div`
  font-weight: bold;
  text-transform: capitalize;
  text-align: right;
  
  &::after {
    content: ':';
  }
`;
const InfoValue = styled.div`
  font-style: italic;
  font-weight: ${props => props.primary ? 'bold' : 'normal'};
  text-transform: ${props => props.primary ? 'uppercase' : 'none'};
`;
const ValueItem = styled.span`
  padding: 5px;
  
  &:not(:first-child) {
    border-left: 1px solid gray;
  }
`;

const Pokemon = ({ data }) => {
  console.log(data);
  return (
    <PokemonWrapper>
      <PokemonImage src={data.thumb} alt={data.name} />
      <PokemonInfo>
        <InfoKey>Name</InfoKey>
        <InfoValue primary>{data.name}</InfoValue>
        <InfoKey>Abilities</InfoKey>
        <InfoValue>{data.abilities.map(_ => <ValueItem key={_}>{_}</ValueItem>)}</InfoValue>
        <InfoKey>Types</InfoKey>
        <InfoValue>{data.types.map(_ => <ValueItem key={_}>{_}</ValueItem>)}</InfoValue>
        <InfoKey>Experience</InfoKey>
        <InfoValue>{data.baseExperience}</InfoValue>
        <InfoKey>Height</InfoKey>
        <InfoValue>{data.height}</InfoValue>
        <InfoKey>Weight</InfoKey>
        <InfoValue>{data.weight}</InfoValue>
        {data.stats.map(_ => (
          <Fragment key={_.name}>
            <InfoKey>{_.name}</InfoKey>
            <InfoValue>{_.base}</InfoValue>
          </Fragment>
        ))}
      </PokemonInfo>
    </PokemonWrapper>
  );
};

export default Pokemon;