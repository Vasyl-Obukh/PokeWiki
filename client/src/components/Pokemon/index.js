import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import * as Styles from './styles';

const Pokemon = ({ data: {thumb = '#', name, abilities, types, baseExperience, height, weight, stats} }) => {
  return (
    <Styles.Wrapper>
      <Styles.Image src={thumb} alt={name} />
      {name && <Styles.Info>
        <Styles.InfoKey>Name</Styles.InfoKey>
        <Styles.InfoValue primary>{name}</Styles.InfoValue>
        <Styles.InfoKey>Abilities</Styles.InfoKey>
        <Styles.InfoValue>{abilities.map(_ => <Styles.ValueItem key={_}>{_}</Styles.ValueItem>)}</Styles.InfoValue>
        <Styles.InfoKey>Types</Styles.InfoKey>
        <Styles.InfoValue>{types.map(_ => <Styles.ValueItem key={_}>{_}</Styles.ValueItem>)}</Styles.InfoValue>
        <Styles.InfoKey>Experience</Styles.InfoKey>
        <Styles.InfoValue>{baseExperience}</Styles.InfoValue>
        <Styles.InfoKey>Height</Styles.InfoKey>
        <Styles.InfoValue>{height}</Styles.InfoValue>
        <Styles.InfoKey>Weight</Styles.InfoKey>
        <Styles.InfoValue>{weight}</Styles.InfoValue>
        {stats.map(_ => (
          <Fragment key={_.name}>
            <Styles.InfoKey>{_.name}</Styles.InfoKey>
            <Styles.InfoValue>{_.base}</Styles.InfoValue>
          </Fragment>
        ))}
      </Styles.Info>}
    </Styles.Wrapper>
  );
};

Pokemon.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string.isRequired,
    abilities: PropTypes.arrayOf(PropTypes.string).isRequired,
    types: PropTypes.arrayOf(PropTypes.string).isRequired,
    stats: PropTypes.arrayOf(PropTypes.object).isRequired,
    thumb: PropTypes.string,
    baseExperience: PropTypes.number,
    height: PropTypes.number,
    weight: PropTypes.number
  })
};

export default Pokemon;