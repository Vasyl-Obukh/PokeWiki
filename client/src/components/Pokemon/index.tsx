import * as React from 'react';
import * as Styles from './styles';

const Pokemon = ({ data: {thumb = '#', name, abilities, types, baseExperience, height, weight, stats} }) => {
  return (
    <Styles.Wrapper>
      <Styles.Image src={thumb} alt={name} />
      {name && <Styles.Info>
        <Styles.InfoKey>Name</Styles.InfoKey>
        <Styles.InfoValue primary>{name}</Styles.InfoValue>
        <Styles.InfoKey>Abilities</Styles.InfoKey>
        <Styles.InfoValue>{abilities.map(_ => <Styles.ValueItem key={_}>{_}</Styles.ValueItem>)}</Styles.InfoValue>
        <Styles.InfoKey>Types</Styles.InfoKey>
        <Styles.InfoValue>{types.map(_ => <Styles.ValueItem key={_}>{_}</Styles.ValueItem>)}</Styles.InfoValue>
        <Styles.InfoKey>Experience</Styles.InfoKey>
        <Styles.InfoValue>{baseExperience}</Styles.InfoValue>
        <Styles.InfoKey>Height</Styles.InfoKey>
        <Styles.InfoValue>{height}</Styles.InfoValue>
        <Styles.InfoKey>Weight</Styles.InfoKey>
        <Styles.InfoValue>{weight}</Styles.InfoValue>
        {stats.map(_ => (
          <React.Fragment key={_.name}>
            <Styles.InfoKey>{_.name}</Styles.InfoKey>
            <Styles.InfoValue>{_.base}</Styles.InfoValue>
          </React.Fragment>
        ))}
      </Styles.Info>}
    </Styles.Wrapper>
  );
};

export default Pokemon;