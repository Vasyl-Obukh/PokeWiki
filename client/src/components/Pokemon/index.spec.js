import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Pokemon from './index';

Enzyme.configure({adapter: new Adapter()});

describe('<Pokemon /> tests', () => {
  let props;

  beforeEach(() => {
    props = {
      data: {
        name: 'pikachu',
        thumb: '/some/url',
        baseExperience: 112,
        height: 5,
        weight: 60,
        types: ['electric'],
        abilities: ['lightning-rod', 'static'],
        stats: [{name: 'hp', base: 35}, {name: 'speed', base: 90}]
      }
    };
  });

  test('Should render correctly with full set of props', () => {
    const component = renderer.create(<Pokemon {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly with missing non-required props', () => {
    delete props.data.thumb;
    delete props.data.baseExperience;
    delete props.data.height;
    delete props.data.weight;

    const component = renderer.create(<Pokemon {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should pass thumb property to img src attribute', () => {
    const component = mount(<Pokemon {...props} />);
    expect(component.find('img').prop('src')).toBe(props.data.thumb);
  });
});