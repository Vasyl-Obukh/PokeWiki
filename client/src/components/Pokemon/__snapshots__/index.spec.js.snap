// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pokemon /> tests Should render correctly with full set of props 1`] = `
<div
  className="sc-bdVaJa gXpakm"
>
  <img
    alt="pikachu"
    className="sc-bwzfXH stlqt"
    src="/some/url"
  />
  <div
    className="sc-htpNat brENIm"
  >
    <div
      className="sc-bxivhb exZLWy"
    >
      Name
    </div>
    <div
      className="sc-ifAKCX iBhDQv"
    >
      pikachu
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      Abilities
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      <span
        className="sc-EHOje fuZplJ"
      >
        lightning-rod
      </span>
      <span
        className="sc-EHOje fuZplJ"
      >
        static
      </span>
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      Types
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      <span
        className="sc-EHOje fuZplJ"
      >
        electric
      </span>
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      Experience
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      112
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      Height
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      5
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      Weight
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      60
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      hp
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      35
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      speed
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      90
    </div>
  </div>
</div>
`;

exports[`<Pokemon /> tests Should render correctly with missing non-required props 1`] = `
<div
  className="sc-bdVaJa gXpakm"
>
  <img
    alt="pikachu"
    className="sc-bwzfXH stlqt"
    src="#"
  />
  <div
    className="sc-htpNat brENIm"
  >
    <div
      className="sc-bxivhb exZLWy"
    >
      Name
    </div>
    <div
      className="sc-ifAKCX iBhDQv"
    >
      pikachu
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      Abilities
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      <span
        className="sc-EHOje fuZplJ"
      >
        lightning-rod
      </span>
      <span
        className="sc-EHOje fuZplJ"
      >
        static
      </span>
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      Types
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      <span
        className="sc-EHOje fuZplJ"
      >
        electric
      </span>
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      Experience
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    />
    <div
      className="sc-bxivhb exZLWy"
    >
      Height
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    />
    <div
      className="sc-bxivhb exZLWy"
    >
      Weight
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    />
    <div
      className="sc-bxivhb exZLWy"
    >
      hp
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      35
    </div>
    <div
      className="sc-bxivhb exZLWy"
    >
      speed
    </div>
    <div
      className="sc-ifAKCX eHjrkF"
    >
      90
    </div>
  </div>
</div>
`;
